packages/@uppy/aws-s3/types/index.test-d.ts

```diff
@@ -1,13 +1,27 @@
 import { Uppy, UppyFile } from '@uppy/core'
-import { expectType } from 'tsd'
+import { expectType, expectError } from 'tsd'
 import AwsS3 from '..'
 
 {
   const uppy = new Uppy()
   uppy.use(AwsS3, {
     getUploadParameters (file) {
       expectType<UppyFile>(file)
-      return { method: 'POST', url: '', fields: {}, headers: {} }
+      return { method: 'POST', url: '' }
+    },
+  })
+  expectError(uppy.use(AwsS3, {
+    shouldUseMultipart: false,
+    getUploadParameters (file) {
+      expectType<UppyFile>(file)
+      return { method: 'POST', url: '' }
+    },
+  }))
+  uppy.use(AwsS3, {
+    shouldUseMultipart: false,
+    getUploadParameters (file) {
+      expectType<UppyFile>(file)
+      return { method: 'POST', url: '', fields: {} }
```

Comment:
Since the changes are quite involved, can we also test the other methods here?

-  getChunkSize
-  createMultipartUpload
-  listParts
-  signPart
-  abortMultipartUpload
-  completeMultipartUpload

Reply:
I'm not sure test coverage for this plugin is that important since we are getting rid of it in the next major. Or we could duplicate the tests from the multipart plugin, but that could easily drift over time and be a pain to maintain. Or maybe we should reduce the coverage of the multipart plugin and focus on this one, with the goal of keeping the tests when switching the modules?

Reply:
I see. I'm not sure where to put it then, but basically just wanted to check if your changes actually work as expected. Just that the functions are allowed, so small addition like:

```js
listParts() {
// ...
}
```

so see if there is no error.